test/ProductConstructionService.ScenarioTests/CodeFlowScenarioTestBase.cs

```diff
@@ -9,34 +12,56 @@ namespace ProductConstructionService.ScenarioTests;
 internal class CodeFlowScenarioTestBase : ScenarioTestBase
 {
     protected async Task CheckForwardFlowGitHubPullRequest(
-        string sourceRepoName,
+        (string Repo, string Commit)[] repoUpdates,
         string targetRepoName,
         string targetBranch,
         string[] testFiles,
         Dictionary<string, string> testFilePatches)
     {
-        PullRequest pullRequest = await WaitForPullRequestAsync(targetRepoName, targetBranch);
+        PullRequest pullRequest = repoUpdates.Length > 1
```

Comment:
this looks a bit weird, but I guess the reason why we're waiting for a PR update here is because we want updates from both repos? Maybe we could add a comment

Reply:
```suggestion
        // When we expect updates from multiple repos (batchable subscriptions), we need to wait until the PR gets updated with the second repository after it is created
        // Otherwise it might try to validate the contents before all updates are in place
        PullRequest pullRequest = repoUpdates.Length > 1
```